### ISSUE TITLE
Polygon2D does not resize when calling resize().

### ISSUE THREAD

**Issue by JosephCatrambone** [godotengine/godot#18573]
**Godot version:**
commit a02fec47a05913318c9b55d7f5f692194d1de5f1
Merge: 4c3c510a8 a1aa98e05

**OS/device including version:**
Windows 10 Pro 64-bit Build 16299 - nVidia GTX 1070

**Issue description:**
Calling resize on the polygon attribute of a Polygon2D does not have an effect.  
Expected behavior: the size of the array should change.

**Steps to reproduce:**
1. Create a Polygon2D with a new GDScript and no verts.
2. Add a child polygon with a few verts -- a square, for example.
3. Attach the following script to the parent.

```
func _ready():
    print(self.polygon == null)
    var num_points = child0.polygon.size()
    print(num_points)
    self.polygon.resize(num_points)
    print(self.polygon.size())
```

Observe that the printed values are 'false', 4, 0.
Should be False, 4, 4.

**Minimal reproduction project:**
[Polygon2DBug.zip](https://github.com/godotengine/godot/files/1969291/Polygon2DBug.zip)

**Possible workaround:**
Create and assign a new PoolVector2Array to the polygon shape.

**Comment by groud** [comment]
Your basic example is a little bit complex for nothing. The same behavior can be achieved using:
```
func _ready():
    self.polygon.resize(6)
    print(self.polygon.size()) # Outputs the current polygon's points numbers, and not 6
```

I'm not sure but this might be an expected behavior. `self.polygon` might return a copy of the PoolVector2Array. Indeed your problem can be workarounded with:
```
func _ready():
    var pol = self.polygon
    pol.resize(6)
    self.polygon = pol
    print(self.polygon.size()) # Outputs 6
```

**Comment by DualMatrix** [comment]
I looked trough the code and yeah it seems intended. .polygon will just return a PoolVector2Array which will Copy on Write afaik.

**Comment by akien-mga** [comment]
This should likely be mentioned in the documentation of the classes which have a `polygon` member:
```
$ rg 'member name="polygon"'
CollisionPolygon2D.xml
25:             <member name="polygon" type="PoolVector2Array" setter="set_polygon" getter="get_polygon">

CollisionPolygon.xml
22:             <member name="polygon" type="PoolVector2Array" setter="set_polygon" getter="get_polygon">

Polygon2D.xml
101:            <member name="polygon" type="PoolVector2Array" setter="set_polygon" getter="get_polygon">

OccluderPolygon2D.xml
22:             <member name="polygon" type="PoolVector2Array" setter="set_polygon" getter="get_polygon">
```

**Comment by JosephCatrambone** [comment]
I added a little bit of documentation to perhaps help people who make the same mistake I did.  I'm open for input on how it should be phrased.

**Comment by akien-mga** [comment]
Documentation needs fixed by #26043.
